pkg/ipam/node.go

```diff
@@ -682,14 +704,25 @@ func (n *Node) maintainIPPool(ctx context.Context) (instanceMutated bool, err er
 		// markedIP can now be considered excess and is not currently in an active handshake
 		ipsToMark = append(ipsToMark, markedIP)
 	}
-	// Update cilium node CRD with IPs that need to be marked for release
+
 	n.mutex.Lock()
 	for _, ip := range ipsToMark {
 		scopedLog.WithFields(logrus.Fields{logfields.IPAddr: ip}).Debug("Marking IP for release")
 		n.ipReleaseStatus[ip] = ipamOption.IPAMMarkForRelease
 	}
 	n.mutex.Unlock()
 
+	// Abort handshake for IPs that are in the middle of handshake, but are no longer considered excess
+	if a.release != nil && len(a.release.IPsToRelease) > 0 {
+		excessMap := make(map[string]bool)
+		for _, ip := range a.release.IPsToRelease {
+			excessMap[ip] = true
+		}
+		n.abortNoLongerExcessIPs(&excessMap)
+	} else {
+		n.abortNoLongerExcessIPs(nil)
+	}
```

Comment:
tiny nit (feel free to ignore): I prefer this form without an else-statement (requires the change mentioned above)

```go
var excessMap map[string]bool
if a.release != nil && len(a.release.IPsToRelease) > 0 {
	excessMap = make(map[string]bool, len(a.release.IPsToRelease))
	for _, ip := range a.release.IPsToRelease {
		excessMap[ip] = true
	}
}
n.abortNoLongerExcessIPs(excessMap)
```

Reply:
Looks much cleaner, updated this.